### PR TITLE
Warn for scripts larger than 1MB compressed

### PR DESCRIPTION
**Pull request by matthewdavidrodgers**
Uses the esbuild-generated metafile to see which dependencies are heaviest and points out the top 5, as well as a message encouraging a smaller script

### CODE REVIEW

File: packages/wrangler/src/bundle-reporter.tsx [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { Blob } from "node:buffer";
 import { gzipSync } from "node:zlib";
 import { logger } from "./logger";
 import type { CfModule } from "./worker";
+import type { Metafile } from "esbuild";
+
+const ONE_MB = 1 << (10 * 2);
```
Comment (JacobMGEvans) [inline packages/wrangler/src/bundle-reporter.tsx:7]: Bit twiddling is not a common pattern in JS projects. Is there a way we can more explicitly represent 1MB here?

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]
Besides the bitwise operator usage, could we have some testing around this too 🙂 

Looks like a great addition though, thanks!

**Comment by matthewdavidrodgers** [comment]
Gonna touch this up today to warn on bigger script sizes, but only report big dependencies if your upload failed due to size restrictions. Will also get a more explicit constant for 1 MB

File: packages/wrangler/src/bundle-reporter.tsx [hunk 2]
```diff
@@ -21,9 +25,44 @@ export async function printBundleSize(
...
 	logger.log(`Total Upload: ${bundleReport}`);
+
+	if (gzipSize > ONE_MIB_BYTES && !process.env.NO_SCRIPT_SIZE_WARNING) {
+		logger.warn(
+			"We recommend keeping your script less than 1MiB (1024 KiB) after gzip. Exceeding past this can affect cold start time"
+		);
+	}
+}
+
+export function printOffendingDependencies(
```
Comment (JacobMGEvans) [inline packages/wrangler/src/bundle-reporter.tsx:41]: Might be something to move to the ESBuild files and export, keeping them co-located. (non-blocking) 

Thoughts @rozenmd ?

**Review by JacobMGEvans — APPROVED** [review]
Tentatively approving. I don't see any blocking issues in my 2nd review! Thanks for this and the requested changes 😄

File: packages/wrangler/src/bundle-reporter.tsx (hunk 2 above)
Comment (matthewdavidrodgers) [reply packages/wrangler/src/bundle-reporter.tsx:41]: yeah, unsure about these failures. some where definitely caused by OOMing node with the big buffers in the tests, but i clamped that down and seems to be better. just rebased off main so hopefully this works

**Pull request merged**